core/shared/src/main/scala/fs2/async/Ref.scala

```diff
@@ -150,9 +153,13 @@ final class Ref[F[_],A] private[fs2] (implicit F: Effect[F], ec: ExecutionContex
       // or the actor directly, and the winner destroys any
       // references behind it!
       if (won.compareAndSet(false, true)) {
-        val actor = ref.get
-        ref.set(null)
-        actor ! Msg.Set(res, () => ())
+        res match {
+          case Left(e) => throw e
```

Comment:
Should the `ref.set(null)` still happen in this case?

Reply:
Good catch! I'll have to think about it

Reply:
Probably yes, or the loser still has a reference, and won't be garbage collected when the winner fails